### PR TITLE
Assorted scaffolder form help text improvements

### PR DESCRIPTION
**Pull request by dtuite**
These commits attempt to give the user hints about what info they should provide when using the scaffolder. These instructions may help them avoid inserting a value which would fail.

### CODE REVIEW

File: plugins/scaffolder-backend/sample-templates/docs-template/template.yaml [hunk 1]
```diff
@@ -21,7 +21,7 @@ spec:
       component_id:
         title: Name
         type: string
-        description: Unique name of the component
+        description: Help others understand what these docs are about.
```
Comment (Rugvip) [inline plugins/scaffolder-backend/sample-templates/docs-template/template.yaml:24]: Think this one is out of place? 😅

File: plugins/scaffolder/src/components/TemplatePage/TemplatePage.tsx [hunk 2]
```diff
@@ -65,7 +65,8 @@ const OWNER_REPO_SCHEMA = {
       format: 'GitHub user or org / Repo name',
       type: 'string' as const,
       title: 'Store path',
-      description: 'GitHub store path in org/repo format',
+      description:
+        'GitHub store path in org/repo format. Scaffolding under personal accounts is currently not supported.',
```
Comment (Rugvip) [inline plugins/scaffolder/src/components/TemplatePage/TemplatePage.tsx:69]: This is about to be fixed by https://github.com/spotify/backstage/pull/2433, so perhaps skip this one?

**Review by Rugvip — COMMENTED** [review]
Thanks!

File: plugins/scaffolder/src/components/TemplatePage/TemplatePage.tsx (hunk 2 above)
Comment (dtuite) [reply plugins/scaffolder/src/components/TemplatePage/TemplatePage.tsx]: Didn't see that. Done.

File: plugins/scaffolder-backend/sample-templates/docs-template/template.yaml (hunk 1 above)
Comment (dtuite) [reply plugins/scaffolder-backend/sample-templates/docs-template/template.yaml]: Oops! Fixed.

**Review by Rugvip — APPROVED** [review]
👍

**Review by stefanalund — APPROVED** [review]

**Pull request merged**